Use pyfakefs to simulate /dev/i2c-1 device in tests

* test_i2c_init_device_not_found
* test_i2c_init_device_permission_error

For example:

```
====================================================================== FAILURES =======================================================================
_____________________________________________________________ test_init_device_not_found ______________________________________________________________

    def test_init_device_not_found():
        from pyfakefs import fake_filesystem
        fs = fake_filesystem.FakeFilesystem()
        os_module = fake_filesystem.FakeOsModule(fs)
        pathname = '/dev/i2c-200'
        new_file = fs.CreateFile(pathname)
    
        port = 200
        address = 0x710
        with patch('smbus2.smbus2.os', os_module):
            with pytest.raises(luma.core.error.DeviceNotFoundError) as ex:
>               i2c(port=port, address=address)

tests/test_i2c.py:43: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
luma/core/serial.py:60: in __init__
    self._bus = bus or smbus2.SMBus(port)
.tox/py35/lib/python3.5/site-packages/smbus2/smbus2.py:132: in __init__
    self.open(bus)
.tox/py35/lib/python3.5/site-packages/smbus2/smbus2.py:142: in open
    self.fd = os.open("/dev/i2c-{}".format(bus), os.O_RDWR)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyfakefs.fake_filesystem.FakeOsModule object at 0x7fb266422cf8>, file_path = '/dev/i2c-200', flags = 2, mode = None

    def open(self, file_path, flags, mode=None):
        """Return the file descriptor for a FakeFile.
    
            WARNING: This implementation only implements creating a file. Please fill
            out the remainder for your needs.
    
            Args:
              file_path: the path to the file
              flags: low-level bits to indicate io operation
              mode: bits to define default permissions
    
            Returns:
              A file descriptor.
    
            Raises:
              OSError: if the path cannot be found
              ValueError: if invalid mode is given
              NotImplementedError: if an unsupported flag is passed in
            """
        if flags & os.O_CREAT:
            fake_file = FakeFileOpen(self.filesystem)(file_path, 'w')
            if mode:
                self.chmod(file_path, mode)
            return fake_file.fileno()
        else:
>           raise NotImplementedError('FakeOsModule.open')
E           NotImplementedError: FakeOsModule.open

.tox/py35/lib/python3.5/site-packages/pyfakefs/fake_filesystem.py:2595: NotImplementedError
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,4 +1,4 @@
-sudo: true
+sudo: false
 language: python
 cache: pip
 
@@ -16,11 +16,6 @@ matrix:
           env: TOXENV=py37
         - python: 2.7
           env: TOXENV=qa
-
-addons:
-  apt:
-    packages:
-    - i2c-tools
 
 install: pip install -U setuptools pip tox coveralls
 script: tox -vv

--- a/setup.py
+++ b/setup.py
@@ -22,7 +22,8 @@ test_deps = [
     'mock;python_version<"3.3"',
     "pytest",
     "pytest-cov",
-    "pytest-warnings"
+    "pytest-warnings",
+    "pyfakefs"
 ]
 
 setup(
@@ -49,7 +50,7 @@ setup(
     tests_require=test_deps,
     extras_require={
         'docs': [
-            'sphinx >= 1.5.1'
+            'sphinx>=1.5.1'
         ],
         'test': test_deps,
         'qa': [

--- a/setup.py
+++ b/setup.py
@@ -22,8 +22,7 @@ test_deps = [
     'mock;python_version<"3.3"',
     "pytest",
     "pytest-cov",
-    "pytest-warnings",
-    "pyfakefs"
+    "pytest-warnings"
 ]
 
 setup(

--- a/tests/test_i2c.py
+++ b/tests/test_i2c.py
@@ -36,7 +36,7 @@ class FakeI2COsModule(fake_filesystem.FakeOsModule):
     def open(self, file_path, flags, mode=None):
         try:
             super(FakeI2COsModule, self).open(file_path, flags, mode)
-        except NotImplementedError as e:
+        except NotImplementedError:
             raise self.expected_error
 
 

--- a/tests/test_i2c.py
+++ b/tests/test_i2c.py
@@ -13,7 +13,10 @@ import smbus2
 from luma.core.serial import i2c
 import luma.core.error
 
+from pyfakefs import fake_filesystem
+
 from helpers import Mock, patch, call
+
 
 smbus = Mock(unsafe=True)
 
@@ -29,23 +32,44 @@ def fib(n):
         a, b = b, a + b
 
 
-def test_init_device_not_found():
+class FakeI2COsModule(fake_filesystem.FakeOsModule):
+    def open(self, file_path, flags, mode=None):
+        try:
+            super(FakeI2COsModule, self).open(file_path, flags, mode)
+        except NotImplementedError as e:
+            raise self.expected_error
+
+
+def test_init_device_not_found(fs):
     port = 200
     address = 0x710
-    with pytest.raises(luma.core.error.DeviceNotFoundError) as ex:
-        i2c(port=port, address=address)
-    assert str(ex.value) == 'I2C device not found: /dev/i2c-{}'.format(port)
+    path_name = '/dev/i2c-{}'.format(port)
+    os_module = FakeI2COsModule(fs)
+    os_module.expected_error = OSError()
+    os_module.expected_error.errno = errno.ENOENT
+    os_module.expected_error.filename = path_name
+
+    with patch('smbus2.smbus2.os', os_module):
+        with pytest.raises(luma.core.error.DeviceNotFoundError) as ex:
+            i2c(port=port, address=address)
+        assert str(ex.value) == 'I2C device not found: {}'.format(path_name)
 
 
-def test_init_device_permission_error():
+def test_init_device_permission_error(fs):
     port = 1
-    try:
-        i2c(port=port)
+    path_name = '/dev/i2c-{}'.format(port)
+    os_module = FakeI2COsModule(fs)
+    os_module.expected_error = OSError()
+    os_module.expected_error.errno = errno.EACCES
+    os_module.expected_error.filename = path_name
 
-    except luma.core.error.DevicePermissionError as ex:
-        # permission error: device exists but no permission
-        assert str(ex) == 'I2C device permission denied: /dev/i2c-{}'.format(
-            port)
+    with patch('smbus2.smbus2.os', os_module):
+        try:
+            i2c(port=port)
+        except luma.core.error.DevicePermissionError as ex:
+            # permission error: device exists but no permission
+            assert str(ex) == 'I2C device permission denied: {}'.format(
+                path_name)
 
 
 def test_init_device_address_error():

--- a/tests/test_i2c.py
+++ b/tests/test_i2c.py
@@ -13,8 +13,6 @@ import smbus2
 from luma.core.serial import i2c
 import luma.core.error
 
-from pyfakefs import fake_filesystem
-
 from helpers import Mock, patch, call
 
 
@@ -32,38 +30,34 @@ def fib(n):
         a, b = b, a + b
 
 
-class FakeI2COsModule(fake_filesystem.FakeOsModule):
-    def open(self, file_path, flags, mode=None):
-        try:
-            super(FakeI2COsModule, self).open(file_path, flags, mode)
-        except NotImplementedError:
-            raise self.expected_error
+def i2c_error(path_name, err_no):
+    expected_error = OSError()
+    expected_error.errno = err_no
+    expected_error.filename = path_name
+
+    def fake_open(a, b):
+        raise expected_error
+    return fake_open
 
 
-def test_init_device_not_found(fs):
+def test_init_device_not_found():
     port = 200
     address = 0x710
     path_name = '/dev/i2c-{}'.format(port)
-    os_module = FakeI2COsModule(fs)
-    os_module.expected_error = OSError()
-    os_module.expected_error.errno = errno.ENOENT
-    os_module.expected_error.filename = path_name
+    fake_open = i2c_error(path_name, errno.ENOENT)
 
-    with patch('smbus2.smbus2.os', os_module):
+    with patch('os.open', fake_open):
         with pytest.raises(luma.core.error.DeviceNotFoundError) as ex:
             i2c(port=port, address=address)
         assert str(ex.value) == 'I2C device not found: {}'.format(path_name)
 
 
-def test_init_device_permission_error(fs):
+def test_init_device_permission_error():
     port = 1
     path_name = '/dev/i2c-{}'.format(port)
-    os_module = FakeI2COsModule(fs)
-    os_module.expected_error = OSError()
-    os_module.expected_error.errno = errno.EACCES
-    os_module.expected_error.filename = path_name
+    fake_open = i2c_error(path_name, errno.EACCES)
 
-    with patch('smbus2.smbus2.os', os_module):
+    with patch('os.open', fake_open):
         try:
             i2c(port=port)
         except luma.core.error.DevicePermissionError as ex:

